pkg/component/kubeproxy/kube_proxy_test.go

```diff
@@ -792,7 +848,12 @@ status: {}
 						"pool-name":          pool.Name,
 						"kubernetes-version": pool.KubernetesVersion.String(),
 					}))
-					Expect(string(managedResourceSecret.Data["daemonset__kube-system__"+daemonSetNameFor(pool)+".yaml"])).To(Equal(daemonSetYAMLFor(pool, values.IPVSEnabled, values.VPAEnabled)))
+
+					if versionutils.ConstraintK8sGreaterEqual129.Check(pool.KubernetesVersion) {
+						Expect(string(managedResourceSecret.Data["daemonset__kube-system__"+daemonSetNameFor(pool)+".yaml"])).To(Equal(daemonSetYAMLFor(pool, values.IPVSEnabled, values.VPAEnabled, true)))
+					} else {
+						Expect(string(managedResourceSecret.Data["daemonset__kube-system__"+daemonSetNameFor(pool)+".yaml"])).To(Equal(daemonSetYAMLFor(pool, values.IPVSEnabled, values.VPAEnabled, false)))
+					}
```

Comment:
Not sure if you want to simplify it, but it could also be written in short:

```suggestion
					Expect(string(managedResourceSecret.Data["daemonset__kube-system__"+daemonSetNameFor(pool)+".yaml"])).To(Equal(daemonSetYAMLFor(pool, values.IPVSEnabled, values.VPAEnabled, versionutils.ConstraintK8sGreaterEqual129.Check(pool.KubernetesVersion))))
```